addOns/custompayloads/src/main/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadsApi.java

```diff
@@ -86,19 +107,125 @@ public ApiResponse handleApiView(String name, JSONObject params) throws ApiExcep
         }
     }
 
-    private List<CustomPayload> getPayloads(String category) throws ApiException {
-        if (!category.isBlank() && !isValidCategory(category)) {
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        LOGGER.debug("handleApiAction  {} {}", name, params);
+        String category;
+        String payload;
+        int[] count = {0};
+
+        switch (name) {
+            case ACTION_DISABLE:
+                category = getValidatedCategory(params.optString(PARAM_CATEGORY));
+                handlePayloads(getAllPayloads(), x -> x.setEnabled(false), category, count);
+                return new ApiResponseElement("disabled", String.valueOf(count[0]));
+
+            case ACTION_ENABLE:
+                category = getValidatedCategory(params.optString(PARAM_CATEGORY));
+                handlePayloads(getAllPayloads(), x -> x.setEnabled(true), category, count);
+                return new ApiResponseElement("enabled", String.valueOf(count[0]));
+
+            case ACTION_REMOVE:
+                category = getValidatedCategory(params.getString(PARAM_CATEGORY));
+                payload = params.getString(PARAM_PAYLOAD);
+                List<CustomPayload> payloads = getAllPayloads();
+                boolean success = false;
+                for (CustomPayload pload : payloads) {
+                    if (pload.getCategory().equals(category)
+                            && pload.getPayload().equals(payload)) {
+                        success = payloads.remove(pload);
+                        extension.getParam().setPayloads(payloads);
+                        break;
+                    }
+                }
+                return success ? ApiResponseElement.OK : ApiResponseElement.FAIL;
+
+            case ACTION_ADD:
+                category = getValidatedCategory(params.getString(PARAM_CATEGORY));
+                payload = params.getString(PARAM_PAYLOAD);
+                if (payload.isBlank()) {
```

Comment:
The GUI allows to add it (and IMO it makes sense to allow it).

Reply:
Note that the `payload` will have to be made optional as empty is considered missing if required.